packages/core/src/sheet.js

```diff
@@ -180,13 +180,11 @@ export const createSheet = (/** @type {DocumentOrShadowRoot} */ root) => {
 }
 
 const addApplyToGroup = (/** @type {RuleGroup} */ group) => {
-	const groupingRule = group.group
-
-	let index = groupingRule.cssRules.length
+	let index = group.group.cssRules.length
 
 	group.apply = (cssText) => {
 		try {
-			groupingRule.insertRule(cssText, index)
+			group.group.insertRule(cssText, index)
```

Comment:
is this differentiation necessary? Unless `group.group` is getting redeclared, which I assume is the case

Reply:
without this we would be evaluating `group.group` to get `groupingRule` too early
we update `group.group` after this point so we need to keep referencing it through `group`